src/colmap/controllers/incremental_pipeline.cc

```diff
@@ -270,6 +284,93 @@ bool IncrementalPipeline::LoadDatabase() {
   return true;
 }
 
+void IncrementalPipeline::Reconstruct(
+    IncrementalMapper& mapper,
+    const IncrementalMapper::Options& mapper_options,
+    bool continue_reconstruction) {
+  for (int num_trials = 0; num_trials < options_->init_num_trials;
```

Comment:
How about splitting this function into 2 levels with one called `ReconstructWithTrials` to reduce the nesting? This will make it more readable and modular.

Reply:
Agreed, but this will again make it hard to review. Let's do this reshuffling of code in another PR and focus here on the logic changes.